include/core/CCompressedDictionary.h

```diff
@@ -125,7 +127,50 @@ class CCompressedDictionary {
         std::string print() const { return CContainerPrinter::print(m_Hash); }
 
     private:
-        TUInt64Array m_Hash;
+        TUInt64Array m_Hash{};
+    };
+
+    //! \brief Translates supported data types to dictionary words.
+    //!
+    //! DESCRIPTION:\n
+    //! We support creating dictionary words for strings, arithmetic types,
+    //! enums and vectors of them.
+    class CTranslator {
+    public:
+        explicit CTranslator(const TUInt64Array& seeds) : m_Hashes{seeds} {}
+
+        void add(const std::string& value) {
+            for (auto& hash : m_Hashes) {
+                hash = CHashing::safeMurmurHash64(
+                    value.c_str(), static_cast<int>(value.size()), hash);
+            }
+        }
+
+        template<typename T>
+        void add(const T& value) {
+            // We don't support POD types because the initialisation
```

Comment:
I do really mean we *shouldn't* support POD types. The problem with the hash implementation uses is reinterpret casts read the objects bytes. So it can read padding bytes. We don't want that.

Reply:
Ok thanks I thought the comment contradicted the code

Reply:
Yes I was really meaning not structs and such, builtin types are fine. I will slightly reword to make this clearer.